### PR TITLE
Fixes #53372 

### LINKED ISSUE #53372: workbench.settings.openDefaultKeybindings && workbench.action.openRawDefaultKeybindings
**New Setting:** `workbench.settings.openDefaultKeybindings`
**New Action:** `workbench.action.openRawDefaultKeybindings`

-----

Currently, `{"workbench.settings.openDefaultSettings": false}` allows users to disable the side-by-side default/user settings view, instead allowing us to view our customized settings file in isolation when using the `workbench.action.openSettings` command.

A corresponding option does not yet exist for `workbench.action.openGlobalKeybindingsFile`, although it would be equally useful.

Additionally, something like @sandy081's `workbench.action.openRawDefaultSettings` contribution in https://github.com/Microsoft/vscode/commit/f8c9ed837ae3306abc55051c44fc32de6e77d6b9 should be implemented to allow access of the default keybindings if `workbench.settings.openDefaultKeybindings` is disabled.

### PR DESCRIPTION
**Pull request by njkevlani**
# Fixes #53372 
### Added following Actions:
- OpenRawDefaultKeybindingsAction
- OpenRawDefaultKeybindingsAction

### Added following Settings:
- openDefaultKeybindings

## What is remaining?
When keybindings are opened with `keyindingsEditor`, it shows User+Default keybindings regardless of what `openDefaultKeybindings` is set to(works for `json` file, though). I wanted to use already implemented `fetch` function, with `@source:user`, for this functionality. But I got some errors. Can you suggest how should I approach this @sandy081?

### CODE REVIEW

**Review by MadanBhandari — APPROVED** [review]

**Comment by njkevlani** [comment]
@sandy081 Done! Review please.

**Comment by njkevlani** [comment]
Just putting a reminder here as there haven't been any activity for a while now. @sandy081

File: src/vs/workbench/parts/preferences/browser/keybindingsEditor.ts [hunk 1]
```diff
@@ -110,6 +110,7 @@ export class KeybindingsEditor extends BaseEditor implements IKeybindingsEditor
 	}
 
 	setInput(input: KeybindingsEditorInput, options: EditorOptions, token: CancellationToken): Thenable<void> {
+		this.searchWidget.setValue(input.defaultSearchValue);
```
Comment (sandy081) [inline src/vs/workbench/parts/preferences/browser/keybindingsEditor.ts]: Why is this change needed?

File: src/vs/workbench/parts/preferences/browser/preferencesActions.ts [hunk 2]
```diff
@@ -125,6 +125,42 @@ export class OpenGlobalKeybindingsFileAction extends Action {
...
+	) {
+		super(id, label);
+	}
+
+	public run(event?: any): TPromise<any> {
+		return this.preferencesService.openRawDefaultKeybindings();
+	}
+}
+
+export class OpenRawUserKeybindingsAction extends Action {
```
Comment (sandy081) [inline src/vs/workbench/parts/preferences/browser/preferencesActions.ts]: We already have an action to open keybindings json file - `OpenGlobalKeybindingsFileAction`. So  we do not need this action.

File: src/vs/workbench/services/preferences/browser/preferencesService.ts [hunk 3]
```diff
@@ -217,23 +221,49 @@ export class PreferencesService extends Disposable implements IPreferencesServic
 				"textual" : { "classification": "SystemMetaData", "purpose": "FeatureInsight", "isMeasurement": true }
 			}
 		*/
+		const openDefaultKeybindings = !!this.configurationService.getValue('workbench.settings.openDefaultKeybindings');
```
Comment (sandy081) [inline src/vs/workbench/services/preferences/browser/preferencesService.ts]: Move this line to inside `if(textual)` block

File: src/vs/workbench/services/preferences/browser/preferencesService.ts [hunk 4]
```diff
@@ -217,23 +221,49 @@ export class PreferencesService extends Disposable implements IPreferencesServic
...
-				return TPromise.join([
-					this.editorService.openEditor({ resource: this.defaultKeybindingsResource, options: { pinned: true, preserveFocus: true }, label: nls.localize('defaultKeybindings', "Default Keybindings"), description: '' }),
-					this.editorService.openEditor({ resource: editableKeybindings, options: { pinned: true } }, sideEditorGroup.id)
-				]).then(editors => void 0);
+				return this.editorService.openEditor({ resource: editableKeybindings, options: { pinned: true } }).then(editors => void 0);
 			});
 		}
-		return this.editorService.openEditor(this.instantiationService.createInstance(KeybindingsEditorInput), { pinned: true }).then(() => null);
+
+		const keybindingsEditorInput = this.instantiationService.createInstance(KeybindingsEditorInput);
```
Comment (sandy081) [inline src/vs/workbench/services/preferences/browser/preferencesService.ts]: I do not think `openDefaultKeybindings` setting should interfere in Keybindings editor. This setting should decide to open or not the default keybindings file when you open keybindings as a file instead of editor.

File: src/vs/workbench/parts/preferences/electron-browser/preferences.contribution.ts [hunk 5]
```diff
@@ -195,6 +195,8 @@ registry.registerWorkbenchAction(new SyncActionDescriptor(OpenSettingsAction, Op
 registry.registerWorkbenchAction(new SyncActionDescriptor(OpenSettings2Action, OpenSettings2Action.ID, OpenSettings2Action.LABEL), 'Preferences: Open Settings (Preview)', category);
 registry.registerWorkbenchAction(new SyncActionDescriptor(OpenGlobalSettingsAction, OpenGlobalSettingsAction.ID, OpenGlobalSettingsAction.LABEL), 'Preferences: Open User Settings', category);
 registry.registerWorkbenchAction(new SyncActionDescriptor(OpenGlobalKeybindingsAction, OpenGlobalKeybindingsAction.ID, OpenGlobalKeybindingsAction.LABEL, { primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyMod.CtrlCmd | KeyCode.KEY_S) }), 'Preferences: Open Keyboard Shortcuts', category);
+registry.registerWorkbenchAction(new SyncActionDescriptor(OpenRawDefaultKeybindingsAction, OpenRawDefaultKeybindingsAction.ID, OpenRawDefaultKeybindingsAction.LABEL), 'Preferences: Open Raw Default Settings', category);
+registry.registerWorkbenchAction(new SyncActionDescriptor(OpenRawUserKeybindingsAction, OpenRawUserKeybindingsAction.ID, OpenRawUserKeybindingsAction.LABEL), 'Preferences: Open Raw Default Settings', category);
```
Comment (sandy081) [inline src/vs/workbench/parts/preferences/electron-browser/preferences.contribution.ts]: Same as previous comment, This is not needed.

File: src/vs/workbench/services/preferences/browser/preferencesService.ts [hunk 6]
```diff
@@ -91,6 +91,10 @@ export class PreferencesService extends Disposable implements IPreferencesServic
 		return this.getEditableSettingsURI(ConfigurationTarget.USER);
 	}
 
+	get userKeybindingsResource(): URI {
```
Comment (sandy081) [inline src/vs/workbench/services/preferences/browser/preferencesService.ts]: Why do we need this?

File: src/vs/workbench/services/preferences/browser/preferencesService.ts [hunk 7]
```diff
@@ -217,23 +221,49 @@ export class PreferencesService extends Disposable implements IPreferencesServic
...
+		}
+
+		return this.editorService.openEditor(keybindingsEditorInput, { pinned: true }).then(() => null);
+	}
+
+	openRawDefaultKeybindings(): TPromise<IEditor> {
+		return this.editorService.openEditor({ resource: this.defaultKeybindingsResource });
+	}
+
+	openRawUserKeybindings(): TPromise<IEditor> {
```
Comment (sandy081) [inline src/vs/workbench/services/preferences/browser/preferencesService.ts]: This is not needed

File: src/vs/workbench/services/preferences/common/preferences.ts [hunk 8]
```diff
@@ -151,6 +151,8 @@ export interface IPreferencesService {
 	openFolderSettings(folder: URI, options?: IEditorOptions, group?: IEditorGroup): TPromise<IEditor>;
 	switchSettings(target: ConfigurationTarget, resource: URI): TPromise<void>;
 	openGlobalKeybindingSettings(textual: boolean): TPromise<void>;
+	openRawDefaultKeybindings(): TPromise<IEditor>;
+	openRawUserKeybindings(): TPromise<IEditor>;
```
Comment (sandy081) [inline src/vs/workbench/services/preferences/common/preferences.ts]: This is not needed

File: src/vs/workbench/services/preferences/common/preferencesEditorInput.ts [hunk 9]
```diff
@@ -77,6 +78,10 @@ export class KeybindingsEditorInput extends EditorInput {
 	matches(otherInput: any): boolean {
 		return otherInput instanceof KeybindingsEditorInput;
 	}
+
+	setDefaultSearchValue(defaultSearchValue = ''): void {
```
Comment (sandy081) [inline src/vs/workbench/services/preferences/common/preferencesEditorInput.ts]: Not needed

**Review by sandy081 — CHANGES_REQUESTED** [review]

**Comment by njkevlani** [comment]
I have pushed requested changes. Review please, @sandy081

File: src/vs/workbench/parts/preferences/electron-browser/preferences.contribution.ts [hunk 10]
```diff
@@ -195,6 +195,7 @@ registry.registerWorkbenchAction(new SyncActionDescriptor(OpenSettingsAction, Op
 registry.registerWorkbenchAction(new SyncActionDescriptor(OpenSettings2Action, OpenSettings2Action.ID, OpenSettings2Action.LABEL), 'Preferences: Open Settings (Preview)', category);
 registry.registerWorkbenchAction(new SyncActionDescriptor(OpenGlobalSettingsAction, OpenGlobalSettingsAction.ID, OpenGlobalSettingsAction.LABEL), 'Preferences: Open User Settings', category);
 registry.registerWorkbenchAction(new SyncActionDescriptor(OpenGlobalKeybindingsAction, OpenGlobalKeybindingsAction.ID, OpenGlobalKeybindingsAction.LABEL, { primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyMod.CtrlCmd | KeyCode.KEY_S) }), 'Preferences: Open Keyboard Shortcuts', category);
+registry.registerWorkbenchAction(new SyncActionDescriptor(OpenRawDefaultKeybindingsAction, OpenRawDefaultKeybindingsAction.ID, OpenRawDefaultKeybindingsAction.LABEL), 'Preferences: Open Raw Default Settings', category);
```
Comment (sandy081) [inline src/vs/workbench/parts/preferences/electron-browser/preferences.contribution.ts]: Label should be `Preferences: Open Default Keyboard Shortcuts File`

**Review by sandy081 — APPROVED** [review]

**Comment by sandy081** [comment]
@njkevlani LGTM. Thanks for the PR

**Comment by njkevlani** [comment]
Thanks for accepting!! @sandy081
